### DISCUSSION TITLE
How to use hash_file as file name?

### DISCUSSION THREAD

**Discussion by jewishmoses** [spatie/laravel-medialibrary#2703] [Q&A]
How would I apply the hash of the flie (`hash_file('md5', $file)`) to be used as a file name?

**Comment by systemsolutionweb** [comment]
read documentation, custom file namer

**Reply by jewishmoses** [reply]
I am aware of the custom file namer, but in order to use the hash_file function you also need the file itself, which I can't access in a custom file namer.
